src/babies/babies.service.ts

```diff
@@ -82,9 +91,54 @@ export class BabiesService {
     );
   }
 
-  async removeBaby(id: string) {
+  async updateBaby(
+    parent: AuthUser,
+    id: string,
+    updateBabyRequestDto: UpdateBabyRequestDto,
+  ) {
     const storedBabyEntity = await this.findBabyOrThrow(id);
 
+    await this.checkIfBabyIsChildOfAuthenticatedParentOrThrow(
+      storedBabyEntity,
+      parent,
+    );
```

Comment:
This is a great approach. Policies is a great approach, but for now we'll keep things simpler. There aren't going to be many validations of this kind I believe. If there is an increase in these types of validations, then we should even think of moving to a business object that validates itself, instead of an entity that does nothing at all. Moving from anemic to domain basically. Thanks for the comment